CHANGELOG.md

```diff
@@ -1,5 +1,282 @@
 # Changelog
 
+## v0.12.0
+
+### Enhancements
+
+#### HTTP authentication support ([#955](https://github.com/DevExpress/testcafe/issues/955), [#1109](https://github.com/DevExpress/testcafe/issues/1109))
+
+TestCafe now supports testing webpages protected with HTTP Basic and NTLM authentication.
+
+Use the `httpAuth` function in fixture or test declaration to specify the credentials.
+
+```js
+fixture `My fixture`
+    .page `http://example.com`
+    .httpAuth({
+        username: 'username',
+        password: 'Pa$$word',
+
+        // Optional parameters, can be required for the NTLM authentication.
+        domain: 'CORP-DOMAIN',
+        workstation: 'machine-win10'
+    });
+
+test('Test1', async t => {});          // Logs in as username
+
+test                                   // Logs in as differentUserName
+    .httpAuth({
+        username: 'differentUserName',
+        password: 'differentPa$$word'
+    })
+    ('Test2', async t => {});
+```
+
+#### Built-in CI-friendly way to start and stop the tested web app ([#1047](https://github.com/DevExpress/testcafe/issues/1047))
+
+When launching tests, you can now specify a command that starts the tested application.
+TestCafe will automatically execute this command before running tests and stop the process when tests are finished.
+
+```sh
+testcafe chrome tests/ --app "node server.js"
+```
+
+```js
+runner
+    .startApp('node server.js')
+    .run();
+```
+
+You can also specify how long TestCafe should wait until the tested application initializes (the default is 1 sec).
+
+```sh
+testcafe chrome tests/ --app "node server.js" --app-init-delay 4000
+```
+
+```js
+runner
+    .startApp('node server.js', 4000)
+    .run();
+```
+
+#### Screenshot and window resize actions now work on Linux ([#1117](https://github.com/DevExpress/testcafe/issues/1117))
+
+The `t.takeScreenshot`, `t.resizeWindow`, `t.resizeWindowToFitDevice` and `t.maximizeWindow` actions can now be executed on Linux machines.
+
+#### Adding custom properties to the element state ([#749](https://github.com/DevExpress/testcafe/issues/749))
+
+The state of webpage elements can now be extended with custom properties.
+
+We have added the [addCustomDOMProperties](https://devexpress,github.io/testcafe/documentation/test-api/selecting-page-elements/selectors.html#adding-custom-properties-to-element-state)
```

Comment:
I think we should give links for corresponding documentation topics for other section (e.g. HTTP authentication support)